### DISCUSSION TITLE
Do not override `tsconfig.json` with `suggestedCompilerOptions`

### DISCUSSION THREAD

**Discussion by f15u** [remix-run/remix#3180] [Proposals]
[`suggestedCompilerOptions`](https://github.com/remix-run/remix/blob/c9d83d58bf4dd87970326c9500837deff5bd8551/packages/remix-dev/compiler/utils/tsconfig/write-config-defaults.ts) override our `tsconfig.json` and is pretty annoying, especially because internally we don't want to use `paths`.

As options being suggested and not required, `tsconfig.json` should not be automatically overrided.

Having an option to at least opt-out would be enough.

**Comment by mcansh** [comment]
@FedericoBiccheddu I'm sorry you don't find it useful, I'll remove that suggestion, for now you should be able to set `paths: {}` to opt out.

in the next release, we'll take any `extends` you're doing into account before applying any suggestions. they'll only be applied if the key is missing

**Reply by lifan0127** [reply]
I have a similar need as I would like to set `moduleResolution` to "Node16", but it get overwritten when remix launches. Is there a workaround for now?

**Reply by mcansh** [reply]
not currently, just opened this https://github.com/remix-run/remix/pull/4034 which im sure we can get in before the next release :)
